aws-xray-recorder-sdk-core/src/main/java/com/amazonaws/xray/entities/NoOpSubSegment.java

```diff
@@ -65,12 +79,14 @@ class NoOpSubSegment implements Subsegment {
         this.creator = creator;
         this.shouldPropagate = shouldPropagate;
         parent = parentSegment;
+        name = "";
+        namespace = "";
 
-        this.isSampled = samplingStrategyOverride == SamplingStrategyOverride.DISABLED ?
-                parentSegment.isSampled() :
-                false;
-        this.isRecording = isSampled;
+        this.isSampled = false;
+        this.isRecording = false;
         this.samplingStrategyOverride = samplingStrategyOverride;
+
+        id = creator.getIdGenerator().newEntityId();
```

Comment:
If this `NoOpSubsegment` is never emitted and never propagated, can we set a static id here? We should avoid unnecessary computation if we can.